java/shared/lang/user_input.yml

```diff
@@ -0,0 +1,57 @@
+type: shared
+languages:
+  - java
+patterns:
+  - pattern: $<REQUEST>.$<REQUEST_METHOD>();
+    filters:
+      - variable: REQUEST
+        detection: java_shared_lang_parameter_request
+      - variable: REQUEST_METHOD
+        values:
+          - getCookies
+          - getHeader
+          - getRequestURI
+          - getRequestURL
+          - getAttribute
+          - getInputStream
+          - getQueryString
+          - getParameter
+          - getParameterMap
+          - getParameterNames
+          - getParameterValues
+          - getReader
+          - getHeaderNames
+          - getPart
+          - getParts
+  - pattern: $<SERVLET_REQUEST>.getHeader($<HEADER>)
+    filters:
+    - variable: SERVLET_REQUEST
+      detection: java_shared_lang_parameter_request
+    - variable: HEADER
+      string_regex: \A(?i)(Host|Referer|User-Agent)\z
+auxiliary:
+  - id: java_shared_lang_parameter_request
```

Comment:
Be aware of the changes I'm making here: https://github.com/Bearer/bearer-rules/pull/119/files#diff-ad4717b095a20d89395d0d5af09ecadbfc7cf72a9ab26600628829b13ae6150cR30-R40

Reply:
Ah, thanks - I'll copy your changes to this PR 👍